src/Configuration/src/KubernetesBase/KubernetesProviderBase.cs

```diff
@@ -38,6 +41,15 @@ internal KubernetesProviderBase(IKubernetes kubernetes, KubernetesConfigSourceSe
             CancellationToken = token;
         }
 
+        internal void ProvideRuntimeReplacements(ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory is not null)
+            {
+                Settings.LoggerFactory = loggerFactory;
+                Logger?.LogTrace("Replacing Bootstrapped loggerFactory with actual factory");
```

Comment:
Shouldn't this method replace `Logger` with a new one from the new loggerFactory? I'm not clear on why we're passing the ILoggerFactory around instead of just replacing the loggers - If I'm reading this right, we start up with 2 (total) ILoggers, but when this replacement happens we are changing out a LoggerFactory and only doing anything for the loggers if they weren't previously set

Reply:
The Logger is getter (no setter). Having that be a property with a backed value was causing its value to be captured in the scope of the lambda, which was causing the old value to be used, even though Logger was later reassigned.

Reply:
Does `Logger` get replaced at runtime if a LoggerFactory is provided for the initial startup?

Reply:
Added a test to prove out ..